src/include/platform/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -586,6 +586,59 @@ CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::SetFailSafeArmed(bool v
     return WriteConfigValue(ConfigClass::kConfigKey_FailSafeArmed, val);
 }
 
+template <class ConfigClass>
+CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetDeviceLocation(MutableDeviceLocation & location)
+{
+    uint8_t locationData[app::Clusters::BasicInformation::kMaxDeviceLocationNameLength + sizeof(DeviceLocatioType)];
+    MutableByteSpan locationSpan(locationData);
+
+    size_t outLen = 0;
+    auto err = ReadConfigValueBin(ConfigClass::kConfigKey_DeviceLocation, locationSpan.data(), locationSpan.size(), outLen);
+    if (err == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND)
+    {
+        location.SetNull();
+        return CHIP_NO_ERROR;
+    }
+    ReturnErrorOnFailure(err);
+
+    TLV::TLVReader tlvReader;
+    tlvReader.Init(locationSpan);
+
+    char locationNameBuffer[1] = { 0 };
+    DeviceLocatioType loc({
+        .locationName = CharSpan(locationNameBuffer, 0),
+    });
+
+    ReturnErrorOnFailure(tlvReader.Next(TLV::AnonymousTag()));
+    ReturnErrorOnFailure(loc->Decode(tlvReader));
+
+    // This would not be needed if the Decode methed proprly decodes a null value.
```

Comment:
@bzbarsky-apple related to the slack message, I cannot find a way to decode a null `DeviceLocatioType` from the encoded bytes. This is the solution that I have came up with so far.